meson.build

```diff
@@ -6,31 +6,132 @@ project('SIMDe', 'c', 'cpp',
 cc = meson.get_compiler('c')
 cxx = meson.get_compiler('cpp')
 
-subdir('test')
-
-install_headers(
-    [
-		'simde/hedley.h',
-		'simde/check.h',
-		'simde/debug-trap.h',
-		'simde/simde-arch.h',
-		'simde/simde-common.h',
-
-		'simde/x86/avx2.h',
-		'simde/x86/avx512f.h',
-		'simde/x86/avx512bw.h',
-		'simde/x86/avx512vl.h',
-		'simde/x86/avx512dq.h',
-		'simde/x86/avx512cd.h',
-		'simde/x86/fma.h',
-		'simde/x86/gfni.h',
-		'simde/x86/mmx.h',
-		'simde/x86/sse.h',
-		'simde/x86/sse2.h',
-		'simde/x86/sse3.h',
-		'simde/x86/sse4.1.h',
-		'simde/x86/sse4.2.h',
-		'simde/x86/ssse3.h',
-		'simde/x86/svml.h',
-    ],
+if not meson.is_subproject() and get_option('tests')
```

Comment:
If it's being built as a subproject the tests won't be build, so I'm not sure we need an option here.  It's not like we're building a shared library or something; the *only* thing to build is the tests, so why would anyone want to "build" SIMDe without building the tests?

The only thing I can think of is that you could install the headers without building the tests, but that doesn't seem particularly useful…  Do you need this feature, or is it just the pattern you use for Meson projects?

Reply:
Oh, nevermind, I just realized: Gentoo.  I guess you want to have a simde package which doesn't actually build anything and only installs to cut down on build times.

Reply:
Correct, we need to be able to disable tests. 95% of Gentoo users don't have tests enabled.